text/3841-demote-x86_64-apple-darwin.md

```diff
@@ -0,0 +1,149 @@
+There is no known long-term replacement for these runners.
+
+## A brief timeline
+
+- 2020-06-22: Apple [announced plans][trans] to shift away from the x86\_64 architecture.
+- 2020-12-31: Rust [promoted `aarch64-apple-darwin`][aarch-tier-2] to Tier 2 with host tools.
+- 2023-06-05: Apple [announced the replacement][trans] of the last x86\_64 hardware.
+- 2023-10-02: GitHub [announces public GitHub Actions runners][m1-runners] for Apple silicon.
+- 2024-10-17: Rust [promoted `aarch64-apple-darwin`][aarch-tier-1] to Tier 1.
+- **2025-07-23**: This RFC opened.
+- 2025-09-01: GitHub [will discontinue][macos-13-sunset] providing free macOS x86\_64 runners for public repositories.
+- 2025 (Fall): [macOS 26][tahoe] will be the last macOS to support the x86\_64 architecture.
+- 2027: The [Rosetta 2][trans] compatibility layer will be mostly removed.
+
+[trans]: https://en.wikipedia.org/wiki/Mac_transition_to_Apple_silicon
+[aarch-tier-2]: https://blog.rust-lang.org/2020/12/31/Rust-1.49.0/#64-bit-arm-macos-and-windows-reach-tier-2
+[aarch-tier-1]: https://blog.rust-lang.org/2024/10/17/Rust-1.82.0/#macos-on-64-bit-arm-is-now-tier-1
+[m1-runners]: https://github.blog/changelog/2023-10-02-github-actions-apple-silicon-m1-macos-runners-are-now-available-in-public-beta/
+[macos-13-sunset]: https://github.blog/changelog/2025-07-11-upcoming-changes-to-macos-hosted-runners-macos-latest-migration-and-xcode-support-policy-updates/#macos-13-is-closing-down
+[tahoe]: https://en.wikipedia.org/wiki/MacOS_Tahoe
+
+## `x86_64-apple-darwin` popularity
+
+Looking at the [public download statistics][dl-stats] for the previous month (retrieved on 2025-07-21),
+we can see that `x86_64-apple-darwin` has substantially fewer downloads than `aarch64-apple-darwin`:
+
+[dl-stats]: https://p.datadoghq.com/sb/3a172e20-e9e1-11ed-80e3-da7ad0900002-60425c7cb1b7beb2e8959a305a301c0c?fromUser=false&refresh_mode=sliding&from_ts=1750525313022&to_ts=1753117313022&live=true
+
+### `rustc`
+
+| platform                   | downloads | percentage |
+|----------------------------|----------:|-----------:|
+| `x86_64-unknown-linux-gnu` |   194.38M |     81.58% |
+| `aarch64-apple-darwin`     |     7.15M |      3.00% |
+| `x86_64-apple-darwin`      |     2.74M |      1.15% |
+
+### `std`
+
+| platform                   | downloads | percentage |
+|----------------------------|----------:|-----------:|
+| `x86_64-unknown-linux-gnu` |    95.12M |     66.20% |
+| `aarch64-apple-darwin`     |     4.82M |      3.35% |
+| `x86_64-apple-darwin`      |     2.76M |      1.92% |
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The first release after this RFC is merged will be the last one with Tier 1 support for the `x86_64-apple-darwin` target.
+The release after that will demote the target to Tier 2 with host tools,
+which means it will not be tested by CI.
```

Comment:
FWIW, we do run some tests for non-tier-1-targets, e.g. some i586 targets. We could probably do the same for x86_64-apple-darwin? IOW, the consequence of this RFC is that we stop *promising* that we run all the tests, and we'll drop what is impractical, but we don't have to drop everything immediately. We could, for instance, keep running the library tests under rosetta, which shouldn't take too long.

Reply:
> Or is our plan really to drop all tests entirely when we stop using the free macos-13 runners?

I think the intention is to keep that open as a possibility in case it becomes necessary.

Reply:
Right, that's what I figured. I think this needs to be quite explicit, or people will conclude that we are dropping all tests immediately.

Reply:
> When the RFC is accepted, immediately stop building the tests.

Okay that is not what I expected. I thought we'd continue to run some of the tests, as we do for a whole bunch of other tier 2 targets. Given the use this target still has, removing *all* testing seems excessive. IIUC, we can keep running tests on the newer macos runners with a 3x slowdown, which should make it fairly easy to still keep some test coverage.

Reply:
> I thought we'd continue to run some of the tests, as we do for a whole bunch of other tier 2 targets.

I don't think we can necessarily guarantee that. We will need to do the demotion, *even if* it means no tests are run. *Even if* it might be possible to run a subset of tests under emulation.

Reply:
> I don't think we can necessarily guarantee that

The RFC currently basically suggests that we *guarantee that we don't run any tests*.
That is not the same as *not guaranteeing that we run all the tests*.

Reply:
Yes. It's definitely a possibility, but it's a matter of continuous evaluation.

Reply:
Yes, I would prefer that.